v2/api/apimanagement/v1api20220801/structure.txt

```diff
@@ -449,103 +674,193 @@ github.com/Azure/azure-service-operator/v2/api/apimanagement/v1api20220801
+│   │   │       ├── "Portal"
+│   │   │       ├── "Proxy"
+│   │   │       └── "Scm"
+│   │   ├── NatGatewayState: *Enum (2 values)
+│   │   │   ├── "Disabled"
+│   │   │   └── "Enabled"
+│   │   ├── NotificationSenderEmail: *string
+│   │   ├── PublicIpAddressId: *string
+│   │   ├── PublicNetworkAccess: *Enum (2 values)
+│   │   │   ├── "Disabled"
+│   │   │   └── "Enabled"
+│   │   ├── PublisherEmail: *string
+│   │   ├── PublisherName: *string
+│   │   ├── Restore: *bool
+│   │   ├── VirtualNetworkConfiguration: *Object (1 property)
+│   │   │   └── SubnetResourceId: *string
+│   │   └── VirtualNetworkType: *Enum (3 values)
+│   │       ├── "External"
+│   │       ├── "Internal"
+│   │       └── "None"
+│   ├── Sku: *Object (2 properties)
+│   │   ├── Capacity: *int
+│   │   └── Name: *Enum (6 values)
+│   │       ├── "Basic"
+│   │       ├── "Consumption"
+│   │       ├── "Developer"
+│   │       ├── "Isolated"
+│   │       ├── "Premium"
+│   │       └── "Standard"
+│   ├── Tags: map[string]string
+│   └── Zones: string[]
+├── Service_Subscription_STATUS_ARM: Object (4 properties)
+│   ├── Id: *string
+│   ├── Name: *string
+│   ├── Properties: *Object (11 properties)
+│   │   ├── AllowTracing: *bool
+│   │   ├── CreatedDate: *string
+│   │   ├── DisplayName: *string
+│   │   ├── EndDate: *string
+│   │   ├── ExpirationDate: *string
+│   │   ├── NotificationDate: *string
+│   │   ├── OwnerId: *string
+│   │   ├── Scope: *string
+│   │   ├── StartDate: *string
+│   │   ├── State: *Enum (6 values)
+│   │   │   ├── "active"
+│   │   │   ├── "cancelled"
+│   │   │   ├── "expired"
+│   │   │   ├── "rejected"
+│   │   │   ├── "submitted"
+│   │   │   └── "suspended"
+│   │   └── StateComment: *string
+│   └── Type: *string
+├── Service_Subscription_Spec_ARM: Object (2 properties)
+│   ├── Name: string
+│   └── Properties: *Object (7 properties)
+│       ├── AllowTracing: *bool
+│       ├── DisplayName: *string
+│       ├── OwnerId: *string
+│       ├── PrimaryKey: *string
+│       ├── Scope: *string
+│       ├── SecondaryKey: *string
+│       └── State: *Enum (6 values)
+│           ├── "active"
+│           ├── "cancelled"
+│           ├── "expired"
+│           ├── "rejected"
+│           ├── "submitted"
+│           └── "suspended"
+└── Subscription: Resource
+    ├── Owner: Service
+    ├── Spec: Object (10 properties)
+    │   ├── AllowTracing: *bool
+    │   ├── AzureName: Validated<string> (2 rules)
+    │   │   ├── Rule 0: MaxLength: 256
+    │   │   └── Rule 1: Pattern: "^[^*#&+:<>?]+$"
+    │   ├── DisplayName: Validated<*string> (2 rules)
+    │   │   ├── Rule 0: MaxLength: 100
+    │   │   └── Rule 1: MinLength: 1
+    │   ├── OperatorSpec: *Object (1 property)
+    │   │   └── Secrets: *Object (2 properties)
+    │   │       ├── PrimaryKey: *genruntime.SecretDestination
+    │   │       └── SecondaryKey: *genruntime.SecretDestination
+    │   ├── Owner: *genruntime.KnownResourceReference
+    │   ├── OwnerId: *string
```

Comment:
OwnerId is actual userId internally. You can create a Subscription at a Product/Api or UserId level. If you fill in Scope it is Product/Api. If it's OwnerId then it will be an APIM User. But I have not added the User type yet.

Reply:
Short answer - I believe that this should be a string

Reply:
Made OwnerId (which is called UserId in Terraform) an Arm Reference